app/src/main/java/com/example/smartsight/DocumentReadingScreen.kt

```diff
@@ -1,33 +1,74 @@
 package com.example.smartsight
 
-import androidx.annotation.StringRes
-import androidx.compose.foundation.Image
+import android.content.Context
+import android.speech.tts.TextToSpeech
+import android.util.Log
 import androidx.compose.foundation.background
 import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
 import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
 import androidx.compose.material.icons.Icons
 import androidx.compose.material.icons.filled.Menu
 import androidx.compose.material3.*
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
-import androidx.compose.ui.tooling.preview.Preview
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.navigation.NavController
+import java.util.*
 
-@OptIn(ExperimentalMaterial3Api::class)
+// *** THIS IS THE FIX ***
+// Use a lateinit var to solve the "variable used before initialization" problem.
+private fun initializeTextToSpeech(
+    context: Context,
+    onInitialized: (TextToSpeech) -> Unit
+) {
+    lateinit var textToSpeech: TextToSpeech // 1. Declare as lateinit
+
+    textToSpeech = TextToSpeech(context) { status -> // 2. Assign to it
+        if (status == TextToSpeech.SUCCESS) {
+            // 3. Now you can safely USE it inside the callback
+            val result = textToSpeech.setLanguage(Locale.getDefault())
+            if (result == TextToSpeech.LANG_MISSING_DATA || result == TextToSpeech.LANG_NOT_SUPPORTED) {
+                Log.e("TTS", "The specified language is not supported!")
+            }
+            onInitialized(textToSpeech)
+        } else {
+            Log.e("TTS", "TTS Initialization failed!")
+        }
+    }
+}
 
 @Composable
 fun DocumentReadingScreen(navController: NavController) {
-    Column(modifier = Modifier.fillMaxSize()
-        .background(Color(0xFFEFEFEF))) {
+    val context = LocalContext.current
+    val scriptToRead = stringResource(id = R.string.speechScripts)
+    var tts: TextToSpeech? by remember { mutableStateOf(null) }
```

Comment:
If camel case is followed throughout the code, then I would rather have a variable names textToSpeech rather than just 'tts'